### DISCUSSION TITLE
Adding a `'none'` fallbackStrategy for falling back to non-relative positioning

### DISCUSSION THREAD

**Discussion by Mesoptier** [floating-ui/floating-ui#2793] [Ideas]
Say I'm positioning a modal using the `flip` middleware with a list of fallback placements. The middleware will attempt each placement in order until one fits. If none fit, it will currently use a placement based on the given `fallbackStrategy`.

My problem with this is that the final placement might not actually fit. So in some situations you might want to fall back to an alternative placement that isn't relative to the reference element (e.g. center in the viewport).

#### Current solution

There's ways to achieve this at the moment. For example with a custom middleware to detect final overflow:
```ts
const isOverflowing = (options: DetectOverflowOptions): Middleware => ({
    name: 'isOverflowing',
    async fn(state) {
        const overflow = await detectOverflow(state, options);
        const isOverflowing = overflow[getSide(state.placement)] > 0;
        return { data: { isOverflowing } };
    },
});
```

With this placed at the end of the middleware list, it will detect if the floating element is still overflowing the viewport after positioning. 

We can then use this information to apply custom fallback styles instead of the ones we'd normally apply based on the result of `computePosition()`.

This really feels like a workaround though. And I personally don't think it integrates well with the Floating UI middleware system.

#### Proposed solution

I would like to propose a `'none'` fallback strategy / placement. This would indicate that no relative positioning should be done at all and make most middleware no-op. Some middleware might have a special case for it, e.g. the `size` middleware could compute the available size of the viewport, disregarding the reference element entirely.

---

I can imagine that this would be quite a breaking change though. So maybe someone in the community has a better solution?

**Comment by atomiks** [comment]
I don't really see an issue with the middleware you created and I think it's better than trying to integrate it directly.
